chanrestore.go

```diff
@@ -48,19 +48,18 @@ type chanDBRestorer struct {
 func (c *chanDBRestorer) openChannelShell(backup chanbackup.Single) (
 	*channeldb.ChannelShell, error) {
 
-	// First, we'll also need to obtain the private key for the shachain
-	// root from the encoded public key.
-	//
-	// TODO(roasbeef): now adds req for hardware signers to impl
-	// shachain...
-	privKey, err := c.secretKeys.DerivePrivKey(backup.ShaChainRootDesc)
+	// Derive the shachain root via an ECDH operation on the private key
+	// of ShaChainRootDesc and the remote node's public key.
+	ecdh, err := c.secretKeys.ECDH(backup.ShaChainRootDesc, backup.RemoteNodePub)
```

Comment:
What if we have to restore a backup for a channel that was written before this change? Won't that create an invalid shachain root? I think we need to support the legacy was here as well and error out if `DerivePrivKey` isn't available because `lnd` runs in "readonly wallet mode".
We'll probably require a node to have no channels when turning on "readonly wallet mode" to make sure they don't run into this. But we do have to support both backup schemes for a long time IMO.

Reply:
We always use `BtcWalletKeyRing` which may change in the future.  But for now both schemes are supported, probably just need an integration test to back it up.

Reply:
I don't see how this would still support "old" channels? With that I mean backups for channels that were created before this PR is merged. If we now always do `ECDH`, the derived revocation root will be wrong for "old" channels.

Reply:
https://github.com/lightningnetwork/lnd/blob/64b8aa8f114142b4b1d1e8e1e751c846d1a7f422/keychain/btcwallet.go#L362

Reply:
whoops, guess I thought it was handled for some reason.  How do you think this should be handled?

Reply:
I think we have to look at the `PubKey` field of the `ShaChainRootDesc`. If it's nil, it's the new scheme, if it's set, it's the old scheme. If it's the old scheme, we just derive the private key. The PR that removes the `DerivePrivKey` in the future will need a way to deal with that (for example erroring out with "restoring old channels on a read-only lnd wallet is not supported") or whatever.

Reply:
Yeah you're right. For some reason I thought it was handled. We have a problem then because if we want a completely *external* signer, and they want to restore an old channel, it won't be possible as the external signer would need to know the details of `DerivePrivKey`. But I guess we can handle it? I guess if that is the case, we would have to error out like you said.

Reply:
Yes, when using the completely external signer mode in lnd, we'd need to error out everywhere we still use `DerivePrivKey`, even if it's just a fallback. We just need to advise users (or enforce it somehow) that they need to start a new node if they use that mode to avoid this problem.